File: Tests/AINotebookCoreTests/LocaleDetectionTests.swift
import XCTest
@testable import AINotebookCore

final class LocaleDetectionTests: XCTestCase {
    func testCzechPreferredReturnsCzech() {
        let result = detectInitialLanguage(preferredLanguages: ["cs-CZ", "en-US"])
        XCTAssertEqual(result, .czech)
    }

    func testCzechWithoutRegionReturnsCzech() {
        let result = detectInitialLanguage(preferredLanguages: ["cs"])
        XCTAssertEqual(result, .czech)
    }

    func testEnglishPreferredReturnsEnglish() {
        let result = detectInitialLanguage(preferredLanguages: ["en-US"])
        XCTAssertEqual(result, .english)
    }

    func testUnknownLanguageDefaultsToEnglish() {
        let result = detectInitialLanguage(preferredLanguages: ["ja-JP", "ko-KR"])
        XCTAssertEqual(result, .english)
    }

    func testEmptyDefaultsToEnglish() {
        let result = detectInitialLanguage(preferredLanguages: [])
        XCTAssertEqual(result, .english)
    }

    func testCzechSecondInListStillCountsAsCzech() {
        // Czech anywhere in preferred list wins because the user explicitly
        // chose it; default English is only when no Czech preference exists.
        let result = detectInitialLanguage(preferredLanguages: ["en-US", "cs-CZ"])
        XCTAssertEqual(result, .czech)
    }
}

File: windows/tests/AINotebook.Core.Tests/LocaleDetectionTests.cs
using AINotebook.Core;
using AINotebook.Core.Models;
using Xunit;

namespace AINotebook.Core.Tests;

public class LocaleDetectionTests
{
    [Fact] // testCzechPreferredReturnsCzech
    public void CzechPreferredReturnsCzech() =>
        Assert.Equal(AppLanguage.Czech, LocaleDetection.DetectInitialLanguage(new[] { "cs-CZ", "en-US" }));

    [Fact] // testCzechWithoutRegionReturnsCzech
    public void CzechWithoutRegionReturnsCzech() =>
        Assert.Equal(AppLanguage.Czech, LocaleDetection.DetectInitialLanguage(new[] { "cs" }));

    [Fact] // testEnglishPreferredReturnsEnglish
    public void EnglishPreferredReturnsEnglish() =>
        Assert.Equal(AppLanguage.English, LocaleDetection.DetectInitialLanguage(new[] { "en-US" }));

    [Fact] // testUnknownLanguageDefaultsToEnglish
    public void UnknownLanguageDefaultsToEnglish() =>
        Assert.Equal(AppLanguage.English, LocaleDetection.DetectInitialLanguage(new[] { "ja-JP", "ko-KR" }));

    [Fact] // testEmptyDefaultsToEnglish
    public void EmptyDefaultsToEnglish() =>
        Assert.Equal(AppLanguage.English, LocaleDetection.DetectInitialLanguage(Array.Empty<string>()));

    [Fact] // testCzechSecondInListStillCountsAsCzech (Czech anywhere wins)
    public void CzechSecondInListStillCountsAsCzech() =>
        Assert.Equal(AppLanguage.Czech, LocaleDetection.DetectInitialLanguage(new[] { "en-US", "cs-CZ" }));
}
